### PR TITLE
Implement `snapcraft update` for parts

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1594643

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

File: snapcraft/internal/parts.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+PARTS_URI = 'https://parts.snapcraft.io/v1/parts.yaml'
+
+logging.getLogger("urllib3").setLevel(logging.CRITICAL)
+logger = logging.getLogger(__name__)
+
+
+class _Base:
+
+    def __init__(self):
+        self.parts_dir = os.path.join(BaseDirectory.xdg_data_home, 'snapcraft')
```
Comment (josepht) [inline snapcraft/internal/parts.py]: Is it possible to have this logic in a single place?

Comment (sergiusens) [inline snapcraft/internal/parts.py]: If the duplication is only in the tests I actually prefer that as we would detect the change

**Comment by josepht** [comment]
Looks good to me.

**Comment by sergiusens** [comment]
retest this please

**Comment by elopio** [comment]
integration tests are failing:
======================================================================
ERROR: test_curl_exists (test_parts.PartsTestCase)
test_parts.PartsTestCase.test_curl_exists
----------------------------------------------------------------------
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/tmp/adt-run.4uGBaN/build.j9u/real-tree/integration_tests/test_parts.py", line 44, in test_curl_exists
    with open(self.parts_yaml) as parts_file:
FileNotFoundError: [Errno 2] No such file or directory: '/home/ubuntu/.local/share/snapcraft/parts.yaml'


======================================================================
FAIL: test_update (test_parts.PartsTestCase)
test_parts.PartsTestCase.test_update
----------------------------------------------------------------------
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/tmp/adt-run.4uGBaN/build.j9u/real-tree/integration_tests/test_parts.py", line 37, in test_update
    self.assertTrue(os.path.exists(self.parts_yaml))
  File "/usr/lib/python3/dist-packages/unittest2/case.py", line 702, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true

**Comment by sergiusens** [comment]
@elopio yeah, fix it fast, the environment mocking logic was different than in unit tests and was passing locally here as I do have a real cache enabled already

**Pull request merged**